### PR TITLE
Fixes package name problem and locale problem

### PR DESCRIPTION
**Pull request by gurkanindibay**
Currently, naming strucuture for rpm is as below
citus_13-11.1.0.citus-0.0.git.20220731.5490c85.el7.x86_64.rpm

When you try to install the package as instructed in citusdata download page, it doesn't work since package name does not include version just after citus as below
citus110_13-11.0.5.citus-1.el7.x86_64.rpm

With the new development, package name will be like below
citus110_13-11.1.0.citus-0.0.git.20220731.5490c85.el7.x86_64.rpm

Additionally I found a bug for centos 7 packages related to locale set. Since C.utf8 is not among supported locales for centos 7, the error below appears in the logs
"sh: warning: setlocale: LC_ALL: cannot change locale (C.utf8): No such file or directory"
When I check the commit history, I found out that the intention to add C.utf8 for Centos is just for el/8 , Not for el/7
Commit hash for this change: e864e961d385190786a5db74ac2e5dfc1132d167

### CODE REVIEW

File: scripts/fetch_and_build_rpm [hunk 1]
```diff
@@ -173,38 +173,42 @@ sed -i -E -e "\\|^Source0:|s|https:.*|${tarballurl}|" \
     "${builddir}/${pkgname}.spec"
 
 osname=$(awk '{print $1}' /etc/system-release)
+os_release=$(awk '{print $4}' /etc/system-release)
 if [ "${osname}" == 'Oracle' ]; then
     locale='C'
-elif [ "${osname}" == 'Fedora' ] || [ "${osname}" == 'CentOS' ]; then
+elif [ "${osname}" == 'Fedora' ] || [[ "${osname}" == 'CentOS'  && "${os_release}" == 8* ]]; then
```
Comment (hanefi) [inline scripts/fetch_and_build_rpm]: I guess it is time that we remove all references to Fedora in our scripts. This hurts readability, and there are a lot of leftover code that do not really have any value.

Can we create an issue and track this please?

File: scripts/fetch_and_build_rpm [hunk 2]
```diff
@@ -173,38 +173,42 @@ sed -i -E -e "\\|^Source0:|s|https:.*|${tarballurl}|" \
...
-            fi
-            sed -i -E "1i %global pkginfix ${package_prefix}" "${builddir}/${pkgname}.spec"
-        fi
         ;;
     nightly)
         msg="Nightly package. Built from ${nightlyref} "
         msg+=$(date +'on %l:%M %p (%Z) on %A, %B %Y' | tr -s ' ')
-        LC_ALL=${locale} rpmdev-bumpspec -c "${msg}" "${builddir}/${pkgname}.spec"
+        LC_ALL=${locale}
+        rpmdev-bumpspec -c "${msg}" "${builddir}/${pkgname}.spec"
```
Comment (hanefi) [inline scripts/fetch_and_build_rpm]: I think these two statements are meant to be on the same line.

```suggestion
        LC_ALL=${locale} rpmdev-bumpspec -c "${msg}" "${builddir}/${pkgname}.spec"
```

File: scripts/fetch_and_build_rpm [hunk 3]
```diff
@@ -173,38 +173,42 @@ sed -i -E -e "\\|^Source0:|s|https:.*|${tarballurl}|" \
...
         sed -i -E 's/0.1.git/0.0.git/' "${builddir}/${pkgname}.spec"
+        sed -i -E "1i %global pkginfix ${package_prefix}" "${builddir}/${pkgname}.spec"
         ;;
     *)
         msg="Custom package. Built from ${gitsha:0:7} "
         msg+=$(date +'on %l:%M %p (%Z) on %A, %B %Y' | tr -s ' ')
-        LC_ALL=${locale} rpmdev-bumpspec -c "${msg}" "${builddir}/${pkgname}.spec"
+        LC_ALL=${locale}
+        rpmdev-bumpspec -c "${msg}" "${builddir}/${pkgname}.spec"
```
Comment (hanefi) [inline scripts/fetch_and_build_rpm]: I think these two statements are meant to be on the same line.

```suggestion
        LC_ALL=${locale} rpmdev-bumpspec -c "${msg}" "${builddir}/${pkgname}.spec"
```

File: scripts/fetch_and_build_rpm [hunk 4]
```diff
@@ -173,38 +173,42 @@ sed -i -E -e "\\|^Source0:|s|https:.*|${tarballurl}|" \
     "${builddir}/${pkgname}.spec"
 
 osname=$(awk '{print $1}' /etc/system-release)
+os_release=$(awk '{print $4}' /etc/system-release)
```
Comment (hanefi) [inline scripts/fetch_and_build_rpm:176]: Nit, can we rename this to `osrelease` so we are consistent with the naming of `osname` ?

**Review by hanefi — COMMENTED** [review]
Looks good overall. Will need to verify that the naming is still as expected after the comments are addressed.

**Comment by hanefi** [comment]
Can we also trigger some nightly and regular image builds so that we can see everything works as expected?

File: scripts/fetch_and_build_rpm (hunk 1 above)
Comment (gurkanindibay) [reply scripts/fetch_and_build_rpm]: https://github.com/citusdata/packaging/issues/960

File: scripts/fetch_and_build_rpm (hunk 2 above)
Comment (gurkanindibay) [reply scripts/fetch_and_build_rpm]: You're right fixed as you suggested

File: scripts/fetch_and_build_rpm (hunk 3 above)
Comment (gurkanindibay) [reply scripts/fetch_and_build_rpm]: You're right fixed as you suggested

File: scripts/fetch_and_build_rpm (hunk 4 above)
Comment (gurkanindibay) [reply scripts/fetch_and_build_rpm:176]: Changed osname into os_name

**Comment by gurkanindibay** [comment]
> Can we also trigger some nightly and regular image builds so that we can see everything works as expected?

Images should be created after merge of the PR to be able to see nightlies with the new images
I performed local test

**Comment by gurkanindibay** [comment]
> > Can we also trigger some nightly and regular image builds so that we can see everything works as expected?
> 
> Images should be created after merge of the PR to be able to see nightlies with the new images I performed local test

image

File: scripts/fetch_and_build_rpm [hunk 5]
```diff
@@ -172,34 +172,36 @@ sed -i -E -e "\\|^Source0:|s|https:.*|${tarballurl}|" \
...
+      package_prefix="${infix}"
+  else
+      package_prefix="${infix}_${release_type}"
+  fi
+  sed -i -E "1i %global pkginfix ${package_prefix}" "${builddir}/${pkgname}.spec"
+fi
+
 case "${1}" in
     release)
         # add minor/major version to package name if using fancy versioning
```
Comment (hanefi) [inline scripts/fetch_and_build_rpm]: This comment no longer makes sense. Can we update that with a simple explanation on why we do not have any steps for release?

**Review by hanefi — APPROVED** [review]
All good after we update a stale comment

**Pull request merged**